### PR TITLE
Use XBlockParser to parse map cubes

### PR DESCRIPTION
**Pull request by kOchirasu**
Type safe parsing for xblock files.

See [Maple2.File.Flat](https://github.com/kOchirasu/Maple2.File/tree/master/Maple2.File.Flat) for type hierarchy, which is based off `*.flat` library files from `Exported.m2d`.

### CODE REVIEW

**Review by darkvergus — APPROVED** [review]

**Review by Sparkymod — APPROVED** [review]
I'm impressed with how easy you make it look like you implement very robust and practical systems. Thank you for letting us learn from you.😊

**Review by AlanMorel — APPROVED** [review]

File: GameDataParser/Parsers/MapParser.cs [hunk 1]
```diff
@@ -2,53 +2,32 @@
...
+using Maple2.File.Parser.MapXBlock;
 using Maple2Storage.Types;
 using Maple2Storage.Types.Metadata;
 
 namespace GameDataParser.Parsers
 {
     public class MapParser : Exporter<List<MapMetadata>>
     {
+        private List<MapMetadata> mapsList;
+        private Dictionary<int, string> mapNames;
```
Comment (AngeloTadeucci) [inline GameDataParser/Parsers/MapParser.cs:19]: we use PascalCase and this file needs some formatting to remove empty white spaces

File: GameDataParser/Files/Resources.cs [hunk 2]
```diff
@@ -1,31 +1,22 @@
 using System.Collections.Generic;
 using System.IO;
 using System.IO.MemoryMappedFiles;
```
Comment (AngeloTadeucci) [inline GameDataParser/Files/Resources.cs:3]: unused usings

File: GameDataParser/Parsers/MapEntityParser.cs [hunk 3]
```diff
@@ -4,8 +4,9 @@
 using System.Linq;
 using System.Text.RegularExpressions;
 using System.Xml;
-using GameDataParser.Crypto.Common;
 using GameDataParser.Files;
+using Maple2.File.IO.Crypto.Common;
+using Maple2.File.Parser.MapXBlock;
```
Comment (AngeloTadeucci) [inline GameDataParser/Parsers/MapEntityParser.cs:9]: unused using

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

**Pull request merged**